packages/core/src/utilities/historyMemo/index.ts

```diff
@@ -73,6 +73,12 @@ export class HistoryMemo {
     }
   }
 
+  public undoIf(condition: (item: Memo) => boolean) {
+    if (condition(this.ring[this.position])) {
```

Comment:
You need to check if there is anything at the given position - number of elements in the undo ring should be at least 1.